src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java

```diff
@@ -177,6 +170,17 @@ private void subscribeToConfigChanges() {
         config.lookupTopics(CONFIGURATION_CONFIG_KEY).subscribe(this::configChangeHandler);
     }
 
+    private void onConfigurationChanged() {
+        runtimeConfiguration = new RuntimeConfiguration(getRuntimeConfig());
+
+        try {
+            caConfiguration = CAConfiguration.from(getConfig());
+            certificateManager.updateCAConfiguration(caConfiguration);
+        } catch (URISyntaxException e) {
+            serviceErrored(e);
```

Comment:
we shouldn't be using serviceErrored unless you want to break the component.

Reply:
Got it and if there is and URISyntax we should break the component, given we that is an invalid configuration

Reply:
No, I don't think you do want that. This component is currently written in a way which is impossible to get out of the broken state. If we do want to break, then we must have a way to get out of that bad state. To get out of broken, you must `requestReinstall()`.

Reply:
Breaking when customer provides garbage config is the appropriate thing to do here.

What's the recommended way to "un-break" yourself? Does a deployment not trigger a restart?

Reply:
No, as we've previous discussed, plugins are not automatically restarted. The plugin can choose to reinstall using `requestReinstall()`. It should do this when configuration changes which will un-break it.

Breaking is generally not the right thing to do because it will take down all components which depend on CDA due to CDA shutting down.

Reply:
Dependent components can't function without valid CA configuration. I'm not worried about that.

That's fine. We can request reinstall if config changes while broken. In the meantime, this is an intermediate PR. CDA already transitions to broken and these config knobs aren't exposed yet. I'd recommend fixing in a follow up PR.